mlir-sycl/lib/Transforms/Inliner.cpp

```diff
@@ -220,8 +260,8 @@ class InlineHeuristic {
   /// Returns true if the target is an ancestor of the call and false otherwise.
   static bool isRecursiveCall(const ResolvedCall &ResolvedCall);
 
-  /// Inlining mode (alwaysinline, simple, ...)
-  const sycl::InlineMode InlineMode;
+  /// Returns the number of operations in the target of the supplied call.
+  static int64_t computeCalleeSize(const ResolvedCall &ResolvedCall);
```

Comment:
Note: this is similar to what the Loop unrolling MLIR transformation does to evaluate the size of an unrolled iteration. 
Estimating the size if a function based on the number of instructions is imprecise because some dialect instructions might expand to several instructions in a lower dialect. If needed, we can refine this rough count by a combination of (as an example):
   - excluding certain dialect instructions from the count
   - assigning weights to instructions (i.e. instr. known to expand to large instr. counts might have a large weight)
   - return MAX_LLONG if the callee contains instructions in a 'disallowed list'